### ISSUE TITLE
Flow Expression parser: cannot parse expression referring to static field of outer class

### ISSUE THREAD

**Issue by jthaine** [typetools/checker-framework#764]
When type checking this code with the Nullness Checker:
````
import org.checkerframework.checker.nullness.qual.*;

public class Issue764 {
    public static @Nullable Object field = null;
    static class MyClass {
        @RequiresNonNull("field")
        public static void method() { }

        public void otherMethod() {
            field = new Object();
            method();
        }
        public void otherMethod2() {
            //:: error: (contracts.precondition.not.satisfied)
            method();
        }
    }
````
Expected: `contracts.precondition.not.satisfied` on the call to `method()`
Actual:  no error

 e717441 is a possible fix for this issue, ~~but it can't be fully tested because of issue general dataflow problems with referencing a static field in an inner class (#818).~~

~~Blocked by #818~~
Revert part of this comment that suppresses a warning related to this issue: https://github.com/codespecs/daikon/commit/437ca4a30c8fd8fa2489e26ae38253117640ad1c

**Comment by mernst** [comment]
I added a test case in file `checker-framework/checker/tests/nullness/Issue764.java`, but I cannot reproduce the parse failure.  Compilation did not produce any warnings or errors, though, and that is a bug.

**Comment by jthaine** [comment]
Thanks for adding the test case. The failure reproduces in branch issue657 (I updated the description above to mention that).

**Comment by jthaine** [comment]
**Important**: this is only closed in branch issue657.

**Comment by smillst** [comment]
I've improved the test case so that it fails on master. (checker/tests/nullness/Issue764.java)  Since this issue it not yet fixed on master, I'm re-opening it.

**Comment by smillst** [comment]
See e717441 from a possible fix for this issue
